src/shadowfax.py

```diff
@@ -220,217 +240,73 @@ async def shadowfax(message, guid, logger):
-            sha256 = hashlib.sha256()
-            for x in set(after_curie_edges):
-                sha256.update(bytes(x, encoding="utf-8"))
-            after_curie_edges_key = sha256.hexdigest()
-            if after_curie_edges_key not in after_curie_edges_keys:
-                aux_graphs[after_curie_edges_key] = {"edges": list(after_curie_edges), "attributes": []}
-                after_curie_edges_keys.append(after_curie_edges_key)
-        main_edge = uuid.uuid1().hex
-        before_edge = uuid.uuid1().hex
-        after_edge = uuid.uuid1().hex
-        knowledge_graph["edges"][main_edge] = {
-            "subject": source_node,
-            "object": target_node,
-            "predicate": "biolink:related_to",
-            "sources": [
-                {
-                    "resource_id": "infores:aragorn",
-                    "resource_role": "primary_knowledge_source",
-                }
-            ],
-            "attributes": [
-                {
-                    "attribute_type_id": "biolink:support_graphs",
-                    "value": aux_edges_keys
-                },
-                {
-                    "attribute_type_id": "biolink:agent_type",
-                    "value": "computational_model",
-                    "attribute_source": "infores:aragorn"
-                },
-                {
-                    "attribute_type_id": "biolink:knowledge_level",
-                    "value": "prediction",
-                    "attribute_source": "infores:aragorn"
-                }
-            ]
-        }
-        knowledge_graph["edges"][before_edge] = {
-            "subject": source_node,
-            "object": curie,
-            "predicate": "biolink:related_to",
-            "sources": [
-                {
-                    "resource_id": "infores:aragorn",
-                    "resource_role": "primary_knowledge_source",
-                }
-            ],
-            "attributes": [
-                {
-                    "attribute_type_id": "biolink:support_graphs",
-                    "value": before_curie_edges_keys
-                },
-                {
-                    "attribute_type_id": "biolink:agent_type",
-                    "value": "computational_model",
-                    "attribute_source": "infores:aragorn"
-                },
-                {
-                    "attribute_type_id": "biolink:knowledge_level",
-                    "value": "prediction",
-                    "attribute_source": "infores:aragorn"
-                }
-            ]
-        }
-        knowledge_graph["edges"][after_edge] = {
-            "subject": curie,
-            "object": target_node,
-            "predicate": "biolink:related_to",
-            "sources": [
+            else:
+                logger.info(f"{guid}: Additional node in pathfinder query.")
+                return message, 500
+
+    result = {
+        "node_bindings": {
+            source_node_key: [
```

Comment:
`source_node_key` and `target_node_key` are possibly unbound. Please ensure you're using an adequate linter (the CoVar one is a good place to start)

Reply:
I don't agree with how this is just making the linter happy. All this is doing is now if there is an issue with these variables, the result will have results that have empty strings as the keys, which is arguably worse than erroring out because it would take someone potentially manually looking at the results to notice. If we can't get the actual node keys to put into the result, then we need to throw an error.

Looking at what the code is trying to do, why don't we set the node keys closer to where we set the node curies? It seems to me like the keys could get set as soon as the pinned_node_ids list gets made. Source node is always the first one and target node is always the second one.

Reply:
Yeah, I think that's fair, less from a logic or code safety point of view, since either way, it would be impossible for those variables to be left unassigned. It's not really any worse, since they could not have those empty strings as keys. We would've already thrown the error and never gotten to that point.

But you make a good point, just from a legibility point of view, it becomes easier to read if we do it that way.